Feature: dependency updates

mikheevm opened the pull request:
## What it solves
Updates the majority of dependencies, some of them require too much effort and are not worth it (immutable, connected-react-router) 

It's based on https://github.com/gnosis/safe-react/pull/2149 because I'm on M1, it needs to be merged first

katspaugh commented:
The jobs fail with `lockfile needs to be updated`.

katspaugh commented:
Also is the base branch of this PR correct?

mikheevm commented:
> The jobs fail with `lockfile needs to be updated`.

Interesting, I'm not able to see it at all. Apparently they're degraded performance https://www.githubstatus.com/

mikheevm commented:
> Also is the base branch of this PR correct?

Depends what you want to review first

katspaugh commented:
> Depends on what you want to review first, I'm not super sure about that tbh

I mean, if it's based on the desktop app update PR, we could just merge this PR into that branch.

mikheevm commented:
> > Depends on what you want to review first, I'm not super sure about that tbh
> 
> I mean, if it's based on the desktop app update PR, we could just merge this PR into that branch.

Then it means this PR has to be reviewed first, isn't it?

katspaugh commented:
Now that I see only the relevant changes, I can review it :)

katspaugh commented on package.json:
```diff
@@ -50,13 +50,8 @@
-  "husky": {
-    "hooks": {
-      "pre-commit": "lint-staged --allow-empty",
-      "pre-push": "tsc --noEmit --incremental"
```
This doesn't work anymore?

mikheevm replied:
it's now inside `.husky` folder

katspaugh replied:
I saw it, just wondering if we could keep it in the json.

mikheevm replied:
I like it outside, personally not a big fan of putting everything into package.json

katspaugh commented on package.json:
```diff
@@ -292,15 +287,15 @@
     "eslint-plugin-react-hooks": "^4.2.0",
-    "husky": "~4.3.8",
-    "lint-staged": "^10.5.2",
+    "husky": "^7.0.0",
```
Wow, that's a huge leap.

dasanra replied:
In less than 6 months, they broke the lib in several majors because they changed it a lot

katspaugh commented on package.json:
```diff
@@ -222,10 +217,10 @@
-    "query-string": "6.14.1",
-    "react": "16.13.1",
+    "query-string": "7.0.1",
+    "react": "17.0.2",
```
🚀

katspaugh commented on package.json:
```diff
@@ -211,8 +206,8 @@
     "immortal-db": "^1.1.0",
-    "immutable": "^4.0.0-rc.12",
-    "js-cookie": "^2.2.1",
+    "immutable": "4.0.0-rc.12",
```
Why removing the ^? Breaking changes after this version?

mikheevm replied:
They added weird types, for example, the return type for `toJS` is `Record<string, unknown` without a way to overwrite it. The benefits are small and long-term I think we want to get rid of this library at all. It's not even maintained anymore.

katspaugh reviewed (approved):
Great work, just had a couple questions.

mikheevm commented:
@katspaugh could you review the arm one too, please?

mikheevm commented:
I removed @gnosis.pm/safe-contracts dependency because it was used only for types, but we also have types from the safe-deployments package. There's only one place where it didn't work out: upgrading safes. I added a `@ts-expect-error` statement there

dasanra commented:
> I removed @gnosis.pm/safe-contracts dependency because it was used only for types that included sqlite3 dependency indirectly which couldn't be compiled with modern node version. But we also have types from the safe-deployments package. There's only one place where it didn't work out: upgrading safes. I added a `@ts-expect-error` statement there

I was planning to do this on a separate PR, as it requieres some more checks and improve a bit the safe-deployments types generation. I already have some ideas for that.

mikheevm commented:
> > I removed @gnosis.pm/safe-contracts dependency because it was used only for types that included sqlite3 dependency indirectly which couldn't be compiled with modern node version. But we also have types from the safe-deployments package. There's only one place where it didn't work out: upgrading safes. I added a `@ts-expect-error` statement there
> 
> I was planning to do this on a separate PR, as it requieres some more checks and improve a bit the safe-deployments types generation. I already have some ideas for that.

It doesn't compile without these changes. Can you maybe work on it from there?

dasanra commented:
> > > I removed @gnosis.pm/safe-contracts dependency because it was used only for types that included sqlite3 dependency indirectly which couldn't be compiled with modern node version. But we also have types from the safe-deployments package. There's only one place where it didn't work out: upgrading safes. I added a `@ts-expect-error` statement there
> > 
> > 
> > I was planning to do this on a separate PR, as it requieres some more checks and improve a bit the safe-deployments types generation. I already have some ideas for that.
> 
> It doesn't compile without these changes. Can you maybe work on it from there?

When I read this PR I was thinking that there was an incompatibility to upgrade storybook to v6 but I wasn't remebering why. Now I remembered and it also motivated my previous removal of truffle dependency. The truffle version included in safe-contracts has some incompatibilities woth storybook 6. But I'm fine keeping it like this :+1:
